### PR TITLE
Add config property to disable new Redshift type mappings

### PR DESCRIPTION
**Pull request by dain**
## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(X) Release notes are required, with the following suggested text:

```markdown
# Redshift
* Implement full type mapping, which can be disable with the `redshift.use-legacy-type-mapping` configuration property ({issue}`15365`)
```

### CODE REVIEW

**Review by martint — APPROVED** [review]

**Review by ebyhr — COMMENTED** [review]
Could you share the manual test result?

**Comment by dain** [comment]
I ran them and they passed

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java [hunk 1]
```diff
@@ -237,9 +238,11 @@ public RedshiftClient(
             JdbcStatisticsConfig statisticsConfig,
             QueryBuilder queryBuilder,
             IdentifierMapping identifierMapping,
-            RemoteQueryModifier queryModifier)
+            RemoteQueryModifier queryModifier,
```
Comment (skrzypo987) [inline plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:241]: nit: queryModifier is always the last arg in *Client classes

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java [hunk 2]
```diff
@@ -467,6 +470,11 @@ protected void verifyColumnName(DatabaseMetaData databaseMetadata, String column
     @Override
     public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle type)
     {
+        // todo remove this when legacy type mapping is no longer supported
```
Comment (skrzypo987) [inline plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:473]: nit: `todo` -> `TODO`

**Review by skrzypo987 — COMMENTED** [review]
typo in release notes:
"which can be disable" -> "which can be disabled"

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftConfig.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+public class RedshiftConfig
+{
+    private boolean legacyTypeMapping;
+
+    public boolean isLegacyTypeMapping()
+    {
+        return legacyTypeMapping;
+    }
+
+    @Config("redshift.use-legacy-type-mapping")
```
Comment (ssheikin) [inline plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftConfig.java:27]: `use` -> `force`
It has been used even before. Even more, it's used in other connectors, but as a fallback. Here it overrides improved mapping.

**Pull request merged**